pkg/api/customization/clusterroletemplatebinding/validator.go

```diff
@@ -0,0 +1,26 @@
+package clusterroletemplatebinding
+
+import (
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/api/customization/roletemplatebinding"
+	"github.com/rancher/types/client/management/v3"
+	"github.com/rancher/types/config"
+)
+
+func NewValidator(management *config.ScaledContext) types.Validator {
```

Comment:
The way you encapsulated the various parts of this really confused me. After reading it a few times, it works, but i think it is failry confusing and hard to follow.

I think the following would be more clear and easier to follow, but let me know if you disagree

```
type Validate struct {
	RoleTemplateLister v3.RoleTemplateLister
        fieldName                string
}

func (v *Validate) ValidateRoleTemplateBinding(obj interface{}) error { // wahtever the normal validator interface is
    // just rely on v.fieldName to get the correct field from the map
}

func NewCRTBValidator(management *config.ScaledContext) types.Validator {
    return &Validate{
             RoleTemplateLister: management.Management.RoleTemplates("").Controller().Lister(),
             fieldName: client.ClusterRoleTemplateBindingFieldRoleTemplateId,
    }
}
// repeat for PRTB
```

and also, collapse these three packages into a single roletemplatebinding package to avoice package bloat. I also think it would be reasonable to put all this log into a single file as well.